File: sdks/typescript/src/resources/inventory.ts
import { AxiosInstance } from 'axios';
import { StockLevel, Product, StockTransfer, PaginatedResponse } from '../types';

export class Inventory {
    constructor(private client: AxiosInstance) { }

    async getStockLevels(branchId?: string): Promise<StockLevel[]> {
        const params = branchId ? { branch: branchId } : {};
        const response = await this.client.get('/inventory/stock-levels/', { params });
        return response.data;
    }

    async getLowStock(threshold: number = 10): Promise<Product[]> {
        const response = await this.client.get('/inventory/low-stock/', {
            params: { threshold }
        });
        return response.data;
    }

    async createTransfer(data: Partial<StockTransfer>): Promise<StockTransfer> {
        const response = await this.client.post('/stock-transfers/', data);
        return response.data;
    }

    async listTransfers(page: number = 1, status?: string): Promise<PaginatedResponse<StockTransfer>> {
        const params: any = { page };
        if (status) {
            params.status = status;
        }
        const response = await this.client.get('/stock-transfers/', { params });
        return response.data;
    }
}

File: sdks/rust/src/resources/inventory.rs
use crate::Puxbay;
use crate::PuxbayError;
use reqwest::Method;
use serde::Deserialize;

pub struct InventoryResource<'a> {
    pub(crate) client: &'a Puxbay,
}

impl<'a> InventoryResource<'a> {
    pub fn new(client: &'a Puxbay) -> Self {
        Self { client }
    }

    pub async fn get_stock_levels(&self, branch_id: &str) -> Result<Vec<serde_json::Value>, PuxbayError> {
        #[derive(Deserialize)]
        struct Response {
            results: Vec<serde_json::Value>,
        }
        
        let response: Response = self.client.request(
            Method::GET,
            &format!("inventory/stock-levels/?branch={}", branch_id),
            None::<()>,
        ).await?;
        
        Ok(response.results)
    }

    pub async fn get_product_stock(&self, product_id: &str, branch_id: &str) -> Result<serde_json::Value, PuxbayError> {
        self.client.request(
            Method::GET,
            &format!("inventory/product-stock/?product={}&branch={}", product_id, branch_id),
            None::<()>,
        ).await
    }
}
